src/NBIHandler.cpp

```diff
@@ -0,0 +1,757 @@
+/**
+ * Implementation of equation term to the heating of cold electrons and ions from an NBI beam
+ */
+#include <cmath>
+#include "DREAM/NBIHandler.hpp"
+#include "FVM/Grid/RadialGrid.hpp"
+#include "DREAM/ADAS.hpp"
+#include "FVM/Grid/Grid.hpp"
+#include "DREAM/Equations/Fluid/IonRateEquation.hpp"
+#include "FVM/UnknownQuantityHandler.hpp"
+#include "FVM/Interpolator1D.hpp"
+#include "DREAM/DREAMException.hpp"
+#include "DREAM/Constants.hpp"
+#include <tuple>
+#include <vector>
+
+
+using namespace DREAM;
+/**
+ * Constructor initializing member variables
+ */
+NBIHandler::NBIHandler(FVM::Grid* grid, ADAS* adas, IonHandler* ions
+): grid(grid), radialGrid(grid->GetRadialGrid()), adas(adas), ions(ions) {
+
+    nr = radialGrid->GetNr();
+    nZ = ions->GetNZ();
+    len_t Zmax_max = 0;
+    for (len_t iz = 0; iz < nZ; ++iz)
+        Zmax_max = std::max(Zmax_max, ions->GetZ(iz));
+    nCharge = Zmax_max + 1;
+    size_t maxidx =
+        static_cast<size_t>(nr) * static_cast<size_t>(nZ) * static_cast<size_t>(nCharge);
+
+    NBIHeatTerm_e = new real_t[nr]();
+    NBIHeatTerm_i = new real_t[nr]();
+    Deposition_profile = new real_t[nr]();
+    Deposition_profile_times_Vprime = new real_t[nr]();
+    dV_beam_prime_tot = new real_t[nr]();
+    H_r_dTe = new real_t[nr]();
+    H_r_dne = new real_t[nr]();
+    d_NBIHeatTerm_e_d_Te = new real_t[nr]();
+    d_NBIHeatTerm_e_d_ne = new real_t[nr]();
+    d_NBIHeatTerm_i_d_Te = new real_t[nr]();
+    d_NBIHeatTerm_i_d_ne = new real_t[nr]();
+    d_NBIHeatTerm_i_d_T_ij = new real_t[maxidx]();
+    d_NBIHeatTerm_i_d_n_ij = new real_t[maxidx]();
+    d_NBIHeatTerm_e_d_T_ij = new real_t[maxidx]();
+    d_NBIHeatTerm_e_d_n_ij = new real_t[maxidx]();
+    H_r_dn_ij = new real_t[maxidx]();
+    H_r_dT_ij = new real_t[maxidx]();
+}
+
+/**
+ * Configure the NBIHandler from NBI settings. Called at the start of the simulation.
+ */
+void NBIHandler::ConfigureFromSettings(
+    Settings* /*s*/, FVM::UnknownQuantityHandler* unknowns, real_t s_max, real_t r_beam,
```

Comment:
If you are not using a parameter in the argument list you can just leave it unnamed. Sometimes we put the name in a comment when the name can help in identifying what the meaning of the parameter is. In this case, the type of ``Settings *s`` is sufficiently clear to not require the variable name to be kept.